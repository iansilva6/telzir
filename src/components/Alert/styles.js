import styled from 'styled-components';

export const Overflow = styled.div`
    position: fixed;
    z-index: 500;
    bottom: 0;
    top: 0%;
    left: 0;
    background: rgba(0, 0, 0, 0.755327);
    height: 100%;
    width: 100%;
`;

export const Alert = styled.div`
    padding: 50px;
    width: 100%;
    border-radius: 26px;
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%,-50%);
    @media (max-width: 768px) {
        padding: 10px;
        border-radius: 10px;
    }
`;

export const Modal = styled.div`
    padding: 50px;
    border-radius: 26px;
    background: #fff;

    img {
        margin: 0 auto;
        display: block;
    }

    h3 {
        font-weight: 600;
        font-size: 24px;
        color: #0BBA94;
        text-align: center;
        margin-top: 12px;
        padding: 10px 0px;
        font-style: normal;
        @media (max-width: 768px) {
            font-size: 20px;
        }
    }

    p {
        font-weight: 300;
        font-size: 16px;
        text-align: center;
        color: #7A7A7A;
        font-style: normal;
        @media (max-width: 768px) {
            font-size: 15px;
        }
    }
`;

export const Buttons = styled.div`
    margin-top: 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    @media (max-width: 768px) {
        margin-top: 15px;
        display: block;
    }
    button {
        text-align: center;
        letter-spacing: 0.573158px;
        text-transform: uppercase;
        color: #000000;
        font-style: normal;
        font-weight: 800;
        font-size: 12px;
        padding: 10px 60px;
        margin: 0px 10px;
        background: #F3F3F3;
        border-radius: 10px;
        border: 2px solid #FFF;
        &:hover {
            border: 2px solid #0BBA94;
            color: #0BBA94;
        }
        @media (max-width: 768px) {
            width: 100%;
            font-size: 12px;
        }
    }
`;