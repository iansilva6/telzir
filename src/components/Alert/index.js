import React from 'react';

// Libs
import { Container, Row, Col } from 'react-bootstrap';

//Styles
import { Overflow, Alert, Modal, Buttons } from './styles';

const Term = (props) => (
    props.show
    ?
    <Overflow>
        <Alert>
            <Container>
                <Row className={"justify-content-center"}>
                    <Col xs={12} sm={10} md={8} lg={6}>
                        <Modal>
                            <img className={"img-responsive"} src={props.image} alt="Atenção"></img>
                            <h3>{props.title}</h3>
                            <p>{props.subtitle}</p>
                            <Buttons>
                                <button onClick={() => props.actionFunction()}>Ok!</button>
                            </Buttons>
                        </Modal>
                    </Col>
                </Row>
            </Container>
        </Alert>
    </Overflow>
    :
    ""
);

export default Term;