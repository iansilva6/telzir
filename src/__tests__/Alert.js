import React from 'react';
import Alert from '../components/Alert';
import renderer from 'react-test-renderer';

// Images and Icons
import DangerImage from '../assets/images/danger.png';

describe("Alert Component", ()=>{
    it("renders as expected", ()=>{
        const tree = renderer
            .create(<Alert
                        show={true}
                        title={"Example Title"}
                        subtitle={"Example Subtitle"}
                        image={DangerImage}
                    />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
});