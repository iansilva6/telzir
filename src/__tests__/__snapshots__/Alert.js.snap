// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component renders as expected 1`] = `
<div
  className="sc-AxjAm cGOBeN"
>
  <div
    className="sc-AxirZ cVjDhq"
  >
    <div
      className="container"
    >
      <div
        className="justify-content-center row"
      >
        <div
          className="col-lg-6 col-md-8 col-sm-10 col-12"
        >
          <div
            className="sc-AxiKw kNkXPg"
          >
            <img
              alt="Atenção"
              className="img-responsive"
              src="danger.png"
            />
            <h3>
              Example Title
            </h3>
            <p>
              Example Subtitle
            </p>
            <div
              className="sc-AxhCb jVvEVs"
            >
              <button
                onClick={[Function]}
              >
                Ok!
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
